File: system-test/typescript/src/testkit/dsl/port/myShop/given/steps/given-order.ts
import type { GivenStage } from '../given-stage.js';
import type { WhenStage } from '../../when/when-stage.js';
import type { ThenStage } from '../../then/then-stage.js';

export interface GivenOrder {
  withOrderNumber(orderNumber: string): GivenOrder;
  withSku(sku: string): GivenOrder;
  withQuantity(quantity: string | number): GivenOrder;
  withCountry(country: string): GivenOrder;
  withCouponCode(couponCode: string | null): GivenOrder;
  withStatus(status: string): GivenOrder;
  and(): GivenStage;
  when(): WhenStage;
  then(): ThenStage;
}

File: system-test/java/src/main/java/com/mycompany/myshop/testkit/dsl/port/given/steps/GivenOrder.java
package com.mycompany.myshop.testkit.dsl.port.given.steps;

import com.mycompany.myshop.testkit.dsl.port.given.steps.base.GivenStep;
import com.mycompany.myshop.testkit.domainvaluetypes.OrderStatus;

public interface GivenOrder extends GivenStep {
    GivenOrder withOrderNumber(String orderNumber);

    GivenOrder withSku(String sku);

    GivenOrder withQuantity(String quantity);

    GivenOrder withQuantity(int quantity);

    GivenOrder withCountry(String country);

    GivenOrder withCouponCode(String couponCode);

    GivenOrder withStatus(String status);

    GivenOrder withStatus(OrderStatus status);
}
